app/src/services/geoStoreService.js

```diff
@@ -152,9 +165,129 @@ class GeoStoreService {
         }
     }
 
+    // @TODO: Extract bbox handling to its own class
+    /**
+     * @name overflowsAntimeridian
+     * @description check if the geometry overflows the [-180, -90, 180, 90] box
+     * @param {Array} bbox
+     * @returns {boolean}
+     */
+    static overflowsAntimeridian(bbox) {
+        return bbox[0] > 180 || bbox[2] > 180;
+    }
+
+    /**
+     * @name bboxToPolygon
+     * @description converts a bbox to a polygon
+     * @param {Array} bbox
+     * @returns {Polygon}
+     */
+    static bboxToPolygon(bbox) {
+        return turf.polygon([[[bbox[2], bbox[3]], [bbox[2], bbox[1]],
+            [bbox[0], bbox[1]], [bbox[0], bbox[3]],
+            [bbox[2], bbox[3]]]]);
+    }
+
+    /**
+     * @name: crossAntiMeridian
+     * @description: checks if a bbox crosses the antimeridian
+     * this is a mirror of https://github.com/mapbox/carmen/blob/03fac2d7397ecdfcb4f0828fcfd9d8a54c845f21/lib/util/bbox.js#L59
+     * @param {Array} bbox A bounding box array in the format [minX, minY, maxX, maxY]
+     * @returns {Array}
+     *
+     */
+    static crossAntimeridian(feature, bbox) {
+        logger.info('Checking antimeridian');
+
+        const geomTypes = ['Point', 'MultiPoint'];
+        const bboxTotal = bbox || turf.bbox(feature);
+        const westHemiBBox = [-180, -90, 0, 90];
+        const eastHemiBBox = [0, -90, 180, 90];
+        const overflowsAntimeridian = this.overflowsAntimeridian(bbox);
+
+        if (geomTypes.includes(turf.getType(feature))) {
+            /**
+             * if the geometry is a triangle geometry length is 4 and
+             * the points are spread among hemispheres bbox calc over each
+             * hemisphere will be wrong
+             * This will need its own development
+             */
+            logger.debug('Multipoint or point geometry');
+            return bboxTotal;
+        }
+
+        if (overflowsAntimeridian) {
+            logger.debug('BBOX crosses antimeridian but is in [0, 360º]');
+            return bboxTotal;
+        }
+
+        if (
+            turf.booleanIntersects(feature, this.bboxToPolygon(eastHemiBBox))
```

Comment:
this is probably also true if the bbox intersects the 0 meridian. is it intentional?

Reply:
yes if the bbox in one of the hemispheres does not cross or only a point crosses it the bbox generated  by turf is infinite and it alters the result of the calc bbox so I put a guard

Same goes on the bboxSplit generated by turf that if is a triangle the bbox cut is a triangle and only one point is out it gives a wrong answer.